lib/pages/common_widgets/goal_widget.dart

```diff
@@ -24,6 +27,24 @@ class GoalWidget extends StatelessWidget {
       });
     }
 
+    doFinishGoal() {
+      final Future<Map<String, dynamic>> responseMessage = finishOrExtend(context, goal.goalId, "true", "0");
+      responseMessage.then((response) {
+        if (response['status'] == true) {
+        onFinish();
+        } else {}
```

Comment:
There should be an alert dialog upon failure

Reply:
done